android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/ResourceExtension.kt

```diff
@@ -38,7 +40,20 @@ fun <T : Resource> T.updateFrom(updatedResource: Resource): T {
   val originalResourceJson = JSONObject(stringJson)
 
   originalResourceJson.updateFrom(JSONObject(updatedResource.toJson(jsonParser)))
-  return jsonParser.parseResource(this::class.java, originalResourceJson.toString())
+  return jsonParser.parseResource(this::class.java, originalResourceJson.toString()).apply {
+    if (this.meta == null || this.meta.isEmpty) {
+      this.meta = this@updateFrom.meta
+      this.meta.tag = this@updateFrom.meta.tag
+    } else {
+      val setOfTags: Set<Coding> = setOf()
+      setOfTags.plus(this.meta.tag)
+      setOfTags.plus(this@updateFrom.meta.tag)
+      this.meta.tag = setOfTags.distinct()
+    }
+    if (this is Patient && this@updateFrom is Patient) {
+      this.extension = this@updateFrom.extension
```

Comment:
Fix this so that we merge both incase new extension are added

Reply:
You can also add tests for this new functionality so that it's caught by the tests if it breaks